### PR TITLE
docs(chart_best_practices): create best practices guide

### LINKED ISSUE #1993: Add basic "Best Practices" guide
We've been meaning to do a best practices doc for a long time. I think we may start with a guide in the `docs` folder here, and gradually build it up over time.

### PR DESCRIPTION
**Pull request by technosophos**
This adds a basic Chart Best Practices guide.

Closes #1993

### CODE REVIEW

**Comment by technosophos** [comment]
The idea here is to at least get something committed that we can continue to add material to as we go.

File: docs/chart_best_practices/labels.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+An item of metadata should be a label under the following conditions:
+
+- It is used by Kubernetes to identify this resource
+- It is useful to expose to operators for the purpose of querying the system.
+
+For example, we suggest using `chart: NAME-VERSION` as a label so that operators
+can conveniently find all of the instances of a particular chart to use.
+
+If an item of metadata is not used for querying, it should be set as a label
```
Comment (wilkers-steve) [inline docs/chart_best_practices/labels.md]: I accidentally left my comments meant for here on your fork.  Should this read "If an item of metadata is not used for querying, it should be set as an annotation instead"?

**Review by wilkers-steve — COMMENTED** [review]
This is a fantastic best practices guide @technosophos.  I feel this sets the stage for future Helm users to see definitive guidelines for any charts they wish to develop publicly.  Great work here!

File: docs/chart_best_practices/labels.md (hunk 1 above)
Comment (bacongobbler) [inline docs/chart_best_practices/labels.md]: kubectl cannot search based on annotations.

```
$ kubectl get pod --all-namespaces --selector=app=helm
```

Comment (wilkers-steve) [inline docs/chart_best_practices/labels.md]: Right.  If its not used for querying, such as your example, should it not be an annotation instead?

File: docs/chart_best_practices/third_party_resources.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+One way to do this is to put the TPR definition in one chart, and then put any
+resources that use that TPR in _another_ chart.
+
+In this method, each chart must be installed separately.
+
+### Method 2: Pre-install Hooks
+
+To package the two together, add a `pre-install` hook to the TPR definition so
+that it is fully installed before the rest of the chart is executed. We strongly
+suggest using the `--wait` flag when installing this way.
```
Comment (thomastaylor312) [inline docs/chart_best_practices/third_party_resources.md]: I don't think the `--wait` flag has any bearing on this. It doesn't wait for hooks, other logic that always runs is responsible for waiting.

**Review by thomastaylor312 — COMMENTED** [review]
One small change. Otherwise this is ready to go. This is an extremely helpful set of documents!

**Comment by technosophos** [comment]
Alright, I made the recommended changes and re-pushed.

**Review by thomastaylor312 — APPROVED** [review]

**Pull request merged**